assets/js/src/marketing_optin_block/block.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/react-in-jsx-scope */
+/* eslint-disable import/no-unresolved */
+
+/**
+ * External dependencies
+ */
+import { CheckboxControl } from '@woocommerce/blocks-checkout';
+import { useState, useEffect } from '@wordpress/element';
+import { getSetting } from '@woocommerce/settings';
+
+const { optinEnabled, defaultText, defaultStatus } = getSetting('mailpoet_data');
+
+const Block = (
+  {
+    text,
+    checkoutExtensionData,
+  }: {
+    text: string,
+    checkoutExtensionData: {
+      setExtensionData: (namespace: string, key: string, value: unknown) => void
+    }
+  }
+): JSX.Element => {
+  const [checked, setChecked] = useState(defaultStatus);
+  const { setExtensionData } = checkoutExtensionData || {};
+  useEffect(() => {
+    if (optinEnabled && setExtensionData) {
+      setExtensionData('mailpoet', 'optin', checked);
+    }
+  }, [checked, setExtensionData]);
+
+  if (!optinEnabled) {
+    return null;
+  }
+
+  return (
+    <CheckboxControl
+      checked={checked}
+      onChange={setChecked}
+    >
+      {/* eslint-disable-next-line react/no-danger */}
```

Comment:
Could you please add a comment on why is this necessary?

Reply:
I had to add this because of the excessive linting rules on commit this repository has implemented. They prohibit the use of dangerouslySet, and in this case, it's needed to support HTML formatting.

Reply:
We added this rule to raise extra attention when using `dangerouslySetInnerHTML`. This may lead to some unexpected security issues. When we need to use it we should provide additional explanations as a code comment for our future selves.

If understand correctly this span is used as a label for the opt-in checkbox and we need to render HTML, so that a user can insert some links etc. Is that correct? Is that safe in terms of XSS? In our [current implementation for old-school checkout](https://github.com/mailpoet/mailpoet/blob/683c5fa01f0fbefeb14d6d8e4fd257303f6a5ffb/lib/WooCommerce/Subscription.php#L84), we use esc_html and we don't allow custom HTML. I understand that it is useful to support links etc. Maybe we could sanitize it by allowing some subset of tags. What about using something like [wp_kses ](https://developer.wordpress.org/reference/functions/wp_kses/)?

Reply:
This content is stored in the post (block attribute) so I believe kses is applied. Adding a note.

> What about using something like wp_kses

This isn't rendered on the PHP side.